Taken from upstream, removed ChangeLog.

From 6930bffe3373690b3431d6291f9f7c116d6a1ec4 Mon Sep 17 00:00:00 2001
From: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date: Sat, 30 May 2020 18:35:59 +0000
Subject: [PATCH] hurd: fix gnu_debug_flag type
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Fixes

../../gdb/gnu-nat.c:96:6: error: conflicting declaration ‘bool gnu_debug_flag’
   96 | bool gnu_debug_flag = false;
../../gdb/gnu-nat.c: In function ‘void _initialize_gnu_nat()’:
../../gdb/gnu-nat.c:3511:7: error: cannot

gdb/ChangeLog:

	* gnu-nat.h (gnu_debug_flag): Set type to bool.
---
 gdb/ChangeLog | 4 ++++
 gdb/gnu-nat.h | 2 +-
 2 files changed, 5 insertions(+), 1 deletion(-)

diff --git a/gdb/gnu-nat.h b/gdb/gnu-nat.h
index 77c57817b2..766f716587 100644
--- a/gdb/gnu-nat.h
+++ b/gdb/gnu-nat.h
@@ -111,7 +111,7 @@ extern char *proc_string (struct proc *proc);
 	      __proc_pid (__proc), __proc->tid, \
 	      host_address_to_string (__proc) , ##args); } while (0)
 
-extern int gnu_debug_flag;
+extern bool gnu_debug_flag;
 
 #define debug(msg, args...) \
  do { if (gnu_debug_flag) \
-- 
Jan Nieuwenhuizen <janneke@gnu.org> | GNU LilyPond http://lilypond.org
Freelance IT http://JoyofSource.com | Avatar® http://AvatarAcademy.com

commit 366f550a593c7e6bae3699a4b6d65fe937af5603
Author: Samuel Thibault <samuel.thibault@ens-lyon.org>
Date:   Sat May 30 18:41:30 2020 +0000

    hurd: add missing include
    
    Fixes
    
    ../../gdb/gnu-nat.c:2522:14: error: ‘target_gdbarch’ was not declared in this scope; did you mean ‘target_detach’?
     2522 |    paddress (target_gdbarch (), memaddr), pulongest (len),
    
    gdb/Changelog:
    
            * gnu-nat.c: Include "gdbarch.h".

diff --git a/gdb/gnu-nat.c b/gdb/gnu-nat.c
index 3b438a9a43..9b93488b41 100644
--- a/gdb/gnu-nat.c
+++ b/gdb/gnu-nat.c
@@ -64,6 +64,7 @@ extern "C"
 #include "language.h"
 #include "target.h"
 #include "gdbsupport/gdb_wait.h"
+#include "gdbarch.h"
 #include "gdbcmd.h"
 #include "gdbcore.h"
 #include "gdbthread.h"
